crates/starknet-server/tests/test_advancing_time.rs

```diff
@@ -0,0 +1,186 @@
+pub mod common;
+
+mod advancing_time_tests {
+
+    use std::time::SystemTime;
+    use std::{thread, time};
+
+    use hyper::Body;
+    use serde_json::json;
+
+    use crate::common::devnet::BackgroundDevnet;
+    use crate::common::utils::get_json_body;
+
+    static DUMMY_ADDRESS: u128 = 1;
+    static DUMMY_AMOUNT: u128 = 1;
+
+    #[tokio::test]
+    async fn set_time_in_past() {
```

Comment:
Why are we testing this? This wasn't a feature of devnet-py, right? There we just said that we support setting to future time, and if users wanted to set timestamp to a past time, we make no guarantees for that. And having the tests for this case looks like making some kind of guarantee, which contradicts the caveat note we have in README.md.

Reply:
This test just testing if it's even possible.

There is no guarantee of any kind, this software is shipped as it is. Please don't use it in critical systems, real-time systems, aviation, and medical devices.

Reply:
but we don't want to know if it's possible

Reply:
it's not a promised feature; tests are meant to test only promised features

Reply:
I'm not promising anything, I'm just checking if timestamp manipulation is possible. I'm not sure if it will be considered during contract execution, there is no test that is checking that.